nodes/modifier_change/offset_line.py

```diff
@@ -0,0 +1,210 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+import bpy
+from bpy.props import FloatProperty
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import updateNode
+
+
+class SvOffsetLineNode(bpy.types.Node, SverchCustomTreeNode):
+    """
+    Triggers: Offset Line 2D
+    Tooltip: Offsetting a Line into 2D space
+
+    Only X and Y dimensions of input points will be taken for work.
+    """
+    bl_idname = 'SvOffsetLineNode'
+    bl_label = 'Offset line'
+    bl_icon = 'OUTLINER_OB_EMPTY'
+
+    offset = FloatProperty(
+        name='offset', description='distance of offset',
+        default=0.1,
+        options={'ANIMATABLE'}, update=updateNode)
+
+    def sv_init(self, context):
+        self.inputs.new('VerticesSocket', 'Vers')
+        self.inputs.new('StringsSocket', "Edgs")
+        self.inputs.new('StringsSocket', "Offset").prop_name = 'offset'
+        self.outputs.new('VerticesSocket', 'Vers')
+        self.outputs.new('StringsSocket', "Faces")
+        self.outputs.new('StringsSocket', "OuterEdges")
+        self.outputs.new('StringsSocket', "VersMask")
+
+    def process(self):
+
+        if not ((self.outputs['Vers'].is_linked or self.outputs['Faces'].is_linked
+              or self.outputs['OuterEdges'].is_linked or self.outputs['VersMask'].is_linked) 
+              and (self.inputs['Vers'].is_linked or self.inputs['Edgs'].is_linked)):
+            return
```

Comment:
I would first test if the `input` vers and edges is linked. _Then_ test if there's also useful output links.